src/main/java/gg/uhc/uhc/modules/health/PercentHealthObjectiveModule.java

```diff
@@ -247,6 +248,27 @@ public void on(EntityDamageEvent event) {
         updatePlayer(player, newHealth);
     }
 
+    @EventHandler(priority = EventPriority.MONITOR)
+    public void on(EntityRegainHealthEvent event) {
+        Entity entity = event.getEntity();
+        if (!(entity instanceof Player)) {
+            return;
+        }
+
+        // TODO: Refactor this instead of copy & paste from EntityDamageEvent?
```

Comment:
I think it's OK here, there isn't a suitable superclass for EntityDamageEvent and EntityRegainHealthEvent and the logic is *slightly* different. Just replicate the comments over from the damage event to explain and it's all good.